internal/auth/repository_auth_method.go

```diff
@@ -0,0 +1,230 @@
+		}
+		defer rows.Close()
+		for rows.Next() {
+			var id string
+			if err := rows.Scan(&id); err != nil {
+				return errors.Wrap(ctx, err, op)
+			}
+			deletedAuthMethodIDs = append(deletedAuthMethodIDs, id)
+		}
+		transactionTimestamp, err = r.Now(ctx)
+		return err
+	}); err != nil {
+		return nil, time.Time{}, err
+	}
+	return deletedAuthMethodIDs, transactionTimestamp, nil
+}
+
+func (amr *AuthMethodRepository) queryAuthMethods(ctx context.Context, query string, args []any) ([]AuthMethod, time.Time, error) {
+	const op = "auth.(*AuthMethodRepository).queryAuthMethods"
+
+	var authmethods []AuthMethod
+	var transactionTimestamp time.Time
+	if _, err := amr.writer.DoTx(ctx, db.StdRetryCnt, db.ExpBackoff{}, func(r db.Reader, w db.Writer) error {
+		rows, err := r.Query(ctx, query, args)
+		if err != nil {
+			return err
+		}
+		defer rows.Close()
+		var foundAuthMethods []*AuthMethodListQueryResult
+		for rows.Next() {
+			if err := r.ScanRows(ctx, rows, &foundAuthMethods); err != nil {
+				return err
+			}
+		}
+
+		for _, am := range foundAuthMethods {
+			switch am.Subtype {
+			case "ldap":
+				ccKey := struct {
+					Ct []byte `wrapping:"ct,certificate_key"`
+					Pt []byte `wrapping:"pt,certificate_key"`
+				}{Ct: am.ClientCertificateKey}
+				if am.ClientCertificateKey != nil {
+					ccWrapper, err := amr.kms.GetWrapper(ctx, am.ScopeId, kms.KeyPurposeDatabase, kms.WithKeyId(am.ClientCertificateKeyId))
+					if err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithMsg("failed to get database wrapper for client certificate key"))
+					}
+					if err := structwrapping.UnwrapStruct(ctx, ccWrapper, &ccKey); err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithCode(errors.Decrypt), errors.WithMsg("failed to decrypt client certificate key"))
+					}
+					am.ClientCertificateKey = ccKey.Pt
+				}
+				bindPassword := struct {
+					Ct []byte `wrapping:"ct,password"`
+					Pt []byte `wrapping:"pt,password"`
+				}{Ct: am.BindPassword}
+				if am.BindPassword != nil {
+					bindWrapper, err := amr.kms.GetWrapper(ctx, am.ScopeId, kms.KeyPurposeDatabase, kms.WithKeyId(am.BindKeyId))
+					if err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithMsg("failed to get database wrapper for bind password"))
+					}
+					if err := structwrapping.UnwrapStruct(ctx, bindWrapper, &bindPassword); err != nil {
+						return errors.Wrap(ctx, err, op, errors.WithCode(errors.Decrypt), errors.WithMsg("failed to decrypt bind password"))
+					}
+					am.BindPassword = bindPassword.Pt
+				}
+			case "oidc":
+				databaseWrapper, err := amr.kms.GetWrapper(ctx, am.ScopeId, kms.KeyPurposeDatabase, kms.WithKeyId(am.KeyId))
+				if err != nil {
+					return errors.Wrap(ctx, err, op, errors.WithMsg("unable to get database wrapper"))
+				}
+				if err := structwrapping.UnwrapStruct(ctx, databaseWrapper, am); err != nil {
+					return errors.Wrap(ctx, err, op, errors.WithCode(errors.Decrypt))
+				}
+			}
```

Comment:
Why are we including secrets when listing at all? It's generally the case that we just drop the secrets from the list responses (e.g. see the credential store list lookup skipping the vault token and certificate key, only the HMACs are included: https://github.com/hashicorp/boundary/blob/f19aa9292758312e06c72ae2c6fd9a5db29b1c51/internal/credential/vault/repository_credential_store.go#L271-L289), so I would think we should just remove these fields from the query and struct.

CC @jimlambrt for thoughts

Reply:
I'm curious what @mgaffney thinks.  My opinion: I'm surprised we're returning secrets from an API which aren't HMAC'd

Reply:
I don't think we should ever return secrets from an API.

Reply:
Seems like we still left some stuff in here that we can remove.